extensions/vertx-mdc/runtime/src/main/java/io/quarkus/vertx/mdc/runtime/VertxMDC.java

```diff
@@ -0,0 +1,311 @@
+package io.quarkus.vertx.mdc.runtime;
+
+import static io.quarkus.vertx.core.runtime.context.VertxContextSafetyToggle.setContextSafe;
+import static io.smallrye.common.vertx.VertxContext.getOrCreateDuplicatedContext;
+
+import java.util.HashMap;
+import java.util.Map;
+import java.util.Objects;
+import java.util.concurrent.ConcurrentHashMap;
+import java.util.concurrent.ConcurrentMap;
+
+import org.jboss.logmanager.MDCProvider;
+
+import io.vertx.core.Context;
+import io.vertx.core.Vertx;
+import io.vertx.core.impl.ContextInternal;
+
+public enum VertxMDC implements MDCProvider {
+    INSTANCE;
+
+    final InheritableThreadLocal<Map<String, Object>> inheritableThreadLocalMap = new InheritableThreadLocal<>() {
+        @Override
+        protected Map<String, Object> childValue(Map<String, Object> parentValue) {
+            if (parentValue == null) {
+                return null;
+            }
+            return new ConcurrentHashMap<>(parentValue);
```

Comment:
If this map is thread-local, wouldn't a plain `HashMap` suffice?

Reply:
Yup =]